Mandatory http response status should be 100..1000

sxmichael opened the pull request:
To have same behavior as WinHttpHandler. Related issue: https://github.com/dotnet/corefx/issues/12937

stephentoub commented:
Thanks, @sxmichael.  Could you add a test to go along with this?  I expect it would look something like this:
```C#
        [Theory]
        [InlineData(200)]
        [InlineData(999)]
        ... // etc, with a few codes to check
        public async Task GetAsync_ExpectedStatusCode(int statusCode)
        {
            await LoopbackServer.CreateServerAsync(async (server, url) =>
            {
                using (var client = new HttpClient())
                {
                    Task<HttpResponseMessage> getResponse = client.GetAsync(url);
                    await LoopbackServer.ReadRequestAndSendResponseAsync(server,
                            $"HTTP/1.1 {statusCode}\r\n" +
                            $"Date: {DateTimeOffset.UtcNow:R}\r\n" +
                            "\r\n");
                    using (HttpResponseMessage response = await getResponse)
                    {
                        Assert.Equal(statusCode, (int)response.StatusCode);
                    }
                }
            });
        }
```

stephentoub commented:
cc: @davidsh, @cipop, @ericeil

sxmichael commented:
@stephentoub Sure, thanks for the code! I've added couple more statuses in the middle to be tested, and added test that for statuses 99 and 1000 expects an exception.

stephentoub commented:
LGTM.  Thanks for the fix, @sxmichael.

stephentoub commented:
@dotnet-bot Test Innerloop Ubuntu14.04 Debug Build and Test please (package restore failures)
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)
@dotnet-bot Test Innerloop Linux ARM Emulator Debug Cross Build please (package restore failures)
@dotnet-bot Test Innerloop CentOS7.1 Release Build and Test please (package restore failures)

justinvp commented on src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs:
```diff
@@ -982,6 +982,51 @@ public static IEnumerable<object[]> GetAsync_IPBasedUri_Success_MemberData()
             });
         }
 
+        [Theory]
```
Should both of the new tests be:

```c#
[OuterLoop] // TODO: Issue #11345
```

?

sxmichael replied:
@justinvp @stephentoub sorry, not sure about that. I do see that many tests in theHttpClientHandlerTest do have this attribute. Just let me know if [OuterLoop] should be there and I'll add it. THanks.

sxmichael replied:
Added OuterLoop (since most of the tests already have it). Please let me know if it should better be removed.

sxmichael commented:
@dotnet-bot Test Innerloop Ubuntu14.04 Release Build and Test please (package restore failures)
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)
@dotnet-bot Test Innerloop Linux ARM Emulator Debug Cross Build please (package restore failures)

sxmichael commented:
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)

sxmichael commented:
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)

stephentoub commented:
(@sxmichael, just FYI, we're having some kind of general issue with restoring a specific package(s) that's causing some CI legs to fail frequently. https://github.com/dotnet/corefx/issues/12949)

sxmichael commented:
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)
@stephentoub yes, I've already noticed that. np

stephentoub commented:
> yes, I've already noticed that

Cool. I saw all of your requests to restart legs, so I just wanted to make sure you knew it wasn't specific to this PR.

sxmichael commented:
@dotnet-bot Test Innerloop Linux ARM Emulator Release Cross Build please (package restore failures)

stephentoub commented:
@dotnet-bot Test Outerloop Windows_NT Debug please

stephentoub commented:
@dotnet-bot Test Outerloop OSX Debug please

sxmichael commented:
@dotnet-bot Test Outerloop Windows_NT Debug please

stephentoub commented:
Thanks for the fix, @sxmichael.

sxmichael commented:
thanks @stephentoub for very quick respones and merge!

sxmichael commented:
@stephentoub actually wanted to ask - is there a chance of including this in 1.1.0?..

stephentoub commented:
> actually wanted to ask - is there a chance of including this in 1.1.0?

Can you help us to understand what kind of issues this is causing for you?  We have a high bar right now for changes to the 1.1 release.

sxmichael commented:
We're working with some external services which return non-standard http codes (>= 600), and most of our infrastructure runs on linux. So currently it prevents us from using dotnet core on linux when working with those services.

stephentoub commented:
It's been merged for 1.1. Thanks.

sxmichael commented:
> It's been merged for 1.1. Thanks.
Thank you! Appreciate it very much!
